app/src/layouts/common/sidebar/sidebar.jsx

```diff
@@ -25,66 +25,73 @@ import { LOGIN_PAGE } from 'controllers/pages/constants';
 import { APPLICATION_SETTINGS } from 'common/constants/localStorageKeys';
 import { updateStorageItem, getStorageItem, fetch } from 'common/utils';
 import { URLS } from 'common/urls';
+import { activeModalSelector } from 'controllers/modal';
 import { NOTIFICATION_TYPES, showNotification } from 'controllers/notification';
+import { useOnClickOutside } from 'common/hooks';
+import { useSelector } from 'react-redux';
 import { SupportBlock } from './supportBlock';
 import LogoutIcon from '../img/logout-inline.svg';
-import { UserBlock } from './userBlock';
+import { OrganizationsBlock } from './organizationsBlock';
+import { Navbar } from './navbar';
+import LogoIcon from './img/logo-icon-inline.svg';
+import { UserAvatar } from './userAvatar';
 import styles from './sidebar.scss';
 
 const cx = classNames.bind(styles);
 
-@connect(null, {
-  logout: logoutAction,
-  showNotification,
-})
-export class Sidebar extends Component {
-  static propTypes = {
-    mainBlock: PropTypes.element,
-    logout: PropTypes.func,
-    topSidebarItems: PropTypes.array,
-    bottomSidebarItems: PropTypes.array,
-    showNotification: PropTypes.func.isRequired,
-  };
-  static defaultProps = {
-    mainBlock: null,
-    logout: () => {},
-    topSidebarItems: [],
-    bottomSidebarItems: [],
-  };
-
-  state = {
-    onboardingOptions: [],
-  };
+export const Sidebar = ({ topSidebarItems, bottomSidebarItems }) => {
+  const [onboardingOptions, setOnboardingOptions] = useState([]);
+  const activeModal = useSelector(activeModalSelector)?.id || '';
+  const [isOpenNavbar, setIsOpenNavbar] = useState(false);
+  const asideRef = useRef(null);
+  let asideTimer;
 
-  componentDidMount() {
+  useLayoutEffect(() => {
     const appSettings = getStorageItem(APPLICATION_SETTINGS);
     const shouldRequestOnboarding = !(appSettings && appSettings.shouldRequestOnboarding === false);
     if (shouldRequestOnboarding) {
       fetch(URLS.onboarding())
         .then((res) => {
           if (Array.isArray(res)) {
-            this.setState({
-              onboardingOptions: res.map(({ problem, link }) => ({ label: problem, value: link })),
-            });
+            setOnboardingOptions(res.map(({ problem, link }) => ({ label: problem, value: link })));
```

Comment:
You can remove onboarding block for now as it will be implemented separately.

Reply:
Done.